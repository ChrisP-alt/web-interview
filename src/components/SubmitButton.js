import React from 'react'
import PropTypes from 'prop-types'

import styles from '../styles/components/submitButton.module.scss'

const SubmitButton = ({ onClick }) => (
  <div>
    <button className={styles.button} onClick={onClick} type="button">
      Book
    </button>
  </div>
)

SubmitButton.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default SubmitButton
