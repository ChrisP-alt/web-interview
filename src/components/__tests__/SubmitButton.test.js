import React from 'react'
import { shallow } from 'enzyme'

import SubmitButton from '../SubmitButton'

describe('SubmitButton component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SubmitButton onClick={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls onClick when the button is clicked', () => {
    const onClick = jest.fn()
    const wrapper = shallow(<SubmitButton onClick={onClick} />)
    wrapper.find('button').simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
