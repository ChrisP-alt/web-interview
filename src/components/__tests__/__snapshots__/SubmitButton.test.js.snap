// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton component renders correctly 1`] = `
<div>
  <button
    className="button"
    onClick={[MockFunction]}
    type="button"
  >
    Book
  </button>
</div>
`;
